dns/route53/dns.go

```diff
@@ -36,34 +36,53 @@ func NewSRVManager(client *Client, hostedZoneID string, recordName string, ttl u
 
 // edit provides both add and removal capabilities
 func (s *SRVManager) edit(add bool, srv *net.SRV) error {
-	log.Infof("Looking up SRV resource record set for %s", s.recordName)
-	resourceRecordSet, err := s.client.GetResourceRecordSetByName(s.hostedZoneID, s.recordName, "SRV")
+	recordName := addDotSuffixIfNeeded(s.recordName)
+
+	log.Infof("Looking up SRV resource record set for %s", recordName)
+	currentResourceRecordSet, err := s.client.GetResourceRecordSetByName(s.hostedZoneID, recordName, "SRV")
 	if err != nil {
 		return err
 	}
 
-	if resourceRecordSet == nil {
-		log.Infof("Resource Record set for %s didn't exist, will create", s.recordName)
+	var resourceRecordSet *route53Client.ResourceRecordSet
+
+	if currentResourceRecordSet == nil {
+		log.Infof("Resource Record set for %s didn't exist, will create", recordName)
 		resourceRecordSet = &route53Client.ResourceRecordSet{
 			TTL:  aws.Int64(int64(s.ttl)),
-			Name: aws.String(s.recordName),
+			Name: aws.String(recordName),
 			Type: aws.String("SRV"),
 		}
+	} else {
+		log.Infof("Resource Record set for %s already exist, will reuse", recordName)
+		resourceRecordSet = currentResourceRecordSet
 	}
 
 	newResourceRecords := editResourceRecords(add, resourceRecordSet.ResourceRecords, srv)
 
+	if currentResourceRecordSet == nil && len(newResourceRecords) <= 0 {
+		log.Infof("skipped creation, no record needed.")
+		return nil
+	}
+
 	if !resourceRecordsDiffer(resourceRecordSet.ResourceRecords, newResourceRecords) {
 		log.Infof("skipped update, no change needed.")
 	} else {
-		resourceRecordSet.ResourceRecords = newResourceRecords
+		var action string
+
+		if len(newResourceRecords) <= 0 {
+			action = route53Client.ChangeActionDelete
+		}else{
```

Comment:
```suggestion
		} else {
```

`gofmt -w .` should show this

Reply:
I just addressed something that I sneaked in accidentially - pushed in 23058542fe8823c3ffe8b8ac8ad41983c63a1bca.